tools/build_script_generator/cmake/resources/CMakeLists.txt.in

```diff
@@ -37,25 +41,55 @@ add_library(modm OBJECT
 
 %% endif
 
-add_library(modm_warnings INTERFACE)
-add_library(modm_options INTERFACE)
-add_library(modm_arch_options INTERFACE)
+%% if is_modm
+add_library({{ repo }}_warnings INTERFACE)
+add_library({{ repo }}_options INTERFACE)
+add_library({{ repo }}_arch_options INTERFACE)
 
-modm_target_config_create(modm modm_arch_options modm_options modm_warnings)
+modm_target_config_create({{ repo }} {{ repo }}_arch_options {{ repo }}_options {{ repo }}_warnings)
+%% endif
```

Comment:
This is only generated ` if is_modm`, so `repo == modm`? Should it instead be `if not is_modm`?

Reply:
No, it should only be generated with `repo == modm` to fill `modm_options`, `modm_warnings` and `modm_arch_options` with the necessary compile flags to then propagate them to all linked targets, but it could be simplified. I will change that.